### PR TITLE
refactor and use @guardian/ab-core in commrcial

### PR DESCRIPTION
**Pull request by dskamiotis**
## What does this change?

This removes most of the ab test framework internally within commercial, and replacing it with the @guardian/ab-core
Since we mainly use isInVariantSynchronous method, this will now allow us to the use the @guardian/ab-core equivalent: isUserInVariant.

This PR now initiates tests using the @guardian/ab-core library and has removed any redundant modules used before

In addition some changes to S3 with the JSON in rendering new liveblog-top ad slot [related PR](https://github.com/guardian/dotcom-rendering/pull/11730)
## Why?

When migrating over from frontend we took with it the whole ab test framework. 
We would like to use the @guardian/ab-core library to maintain consistency with dotcom-rendering particularly when using methods to check if the user is in the test. 

## Testing
Tested with `http://localhost:3030/Article/http:/localhost:9000/sport/2023/nov/22/ronnie-osullivan-warns-he-will-quit-snooker-if-he-cannot-play-in-china` using logs to display the state.

We are using  the  `localhost:9000/` domain after `/Article` in the URL, to ensure the capi/config data is coming from "LOCAL".
This was necessary in order to  switch the feature on safely in local `https://frontend.local.dev-gutools.co.uk/` wihout impacting PROD.

| Feature Switch |
|-----|
| Screenshot 2024-07-15 at 11 23 50 |

In addition the manual setting of the `mvtId` in the browser and temporarily setting the `audience` in the mpuWhenNoEpic test to 10%. 
Then testing different vals (1 - 100,000) with odds and evens being in control/variant of the test.

Note: 
adding the param `#ab-mpuWhenNoEpic=variant`, ensuring the `isUserInvariant` accurately detects:
1. The test
2. User being in the variant

### CODE REVIEW

File: src/experiments/ab.ts [hunk 1]
```diff
@@ -1,91 +1,115 @@
...
+			console.log('AB tests error:', error);
+		},
 	});
+
+	const allRunnableTests = ab.allRunnableTests(concurrentTests);
+	ab.trackABTests(allRunnableTests);
+	ab.registerImpressionEvents(allRunnableTests);
+	ab.registerCompleteEvents(allRunnableTests);
+	log('dotcom', 'AB tests initialised');
+	ab.isUserInVariant;
```
Comment (Jakeii) [inline src/experiments/ab.ts:93]: This line isn't doing anything btw
```suggestion
```

File: src/experiments/ab.ts [hunk 2]
```diff
@@ -1,91 +1,114 @@
...
-		);
+const getOphan = () => {
+	const { ophan } = window.guardian;
 
-		registerImpressionEvents(tests);
-		registerCompleteEvents(tests);
-		trackABTests(tests);
+	const record: OphanRecordFunction = (event, callback) => {
+		ophan.record(event, callback);
+		log('dotcom', '🧿 Ophan event recorded:', event);
```
Comment (Jakeii) [inline src/experiments/ab.ts:48]: I think change these logs to commercial as we're in the commercial code

File: src/experiments/ab.ts [hunk 3]
```diff
@@ -1,91 +1,114 @@
...
+		errorReporter: (error) => {
+			console.log('AB tests error:', error);
+		},
 	});
+
+	const allRunnableTests = ab.allRunnableTests(concurrentTests);
+	ab.trackABTests(allRunnableTests);
+	ab.registerImpressionEvents(allRunnableTests);
+	ab.registerCompleteEvents(allRunnableTests);
+	log('dotcom', 'AB tests initialised');
```
Comment (Jakeii) [inline src/experiments/ab.ts:92]: also this log

File: src/experiments/ab.ts [hunk 4]
```diff
@@ -1,91 +1,114 @@
...
-		setParticipationsInLocalStorage({
-			...getParticipationsFromLocalStorage(),
-			...runnableTestsToParticipations(tests),
+	const trackComponentAttention: typeof ophan.trackComponentAttention = (
+		name,
+		el,
+		visibilityThreshold,
+	) => {
+		ophan.trackComponentAttention(name, el, visibilityThreshold);
+		log('dotcom', '🧿 Ophan tracking component attention:', name, {
```
Comment (Jakeii) [inline src/experiments/ab.ts:57]: last one

**Review by Jakeii — APPROVED** [review]
Just one last log then good to go, nice work!

File: src/experiments/ab.ts (hunk 4 above)
Comment (dskamiotis) [reply src/experiments/ab.ts]: good spot!

**Pull request merged**